src/util/unicode.cpp

```diff
@@ -20,6 +19,24 @@ Author: Daniel Kroening, kroening@kroening.com
 
 /*******************************************************************\
 
+Function: is_little_endian_arch
+
+  Inputs:
+
+ Outputs: True if the architecture is little_endian
+
+ Purpose: Determine endianness of the architecture
+
+\*******************************************************************/
+
+bool is_little_endian_arch()
+{
+  uint16_t i=1;
```

Comment:
Yes, uint8_t will be 8 bits, but I'm not entirely sure whether a char-is-16bits architecture would yield the expected answer here. You may have to go for `uint32_t` to make sure endianness affects byte order?